### DISCUSSION TITLE
Property Name was not yet loaded

### DISCUSSION THREAD

**Discussion by yehen12** [pnp/pnpcore#955] [Q&A] [answered]
![1](https://user-images.githubusercontent.com/29863552/186619778-6aca6d58-4a7a-415c-b084-905558a5a1dd.png)
I'm trying to convert my file format from xls to pdf in SharePoint Online using PnP,
the code is same with Samples from https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#converting-files

but it throw an error when convert excel to pdf:

                string documentUrl = $"{context.Uri.PathAndQuery}/Shared Documents/Test Excel.xls";

                // Get a reference to the file
                IFile testDocument = await context.Web.GetFileByServerRelativeUrlAsync(documentUrl, p => p.VroomItemID, p => p.VroomDriveID);

                var pdfContent = await testDocument.ConvertToAsync(new ConvertToOptions { Format = ConvertToFormat.Pdf });

Any clues?

**Comment by jansenbe** [comment, marked as answer]
@yehen12 : can you also load the name property when loading the file, that should fix this. I'll update the code to ensure that Name was loaded before attempting the extension check.

```csharp
IFile testDocument = await context.Web.GetFileByServerRelativeUrlAsync(documentUrl, p => p.Name, p => p.VroomItemID, p => p.VroomDriveID);
```

**Reply by yehen12** [reply]
After load the name, the Error isn't shown, but it shows TIMEOUT.

TaskCanceledException: A task was canceled.
PnP.Core.Services.RetryHandlerBase.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)

TimeoutException: A task was canceled.
PnP.Core.Services.RetryHandlerBase.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)

TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, bool async, bool emitTelemetryStartStop, CancellationToken cancellationToken)

**Reply by jansenbe** [reply]
@yehen12 : it seems the conversion is taking more than the default timeout, you can configure the timeout to be different by setting the TimeOut property (see https://pnp.github.io/pnpcore/using-the-sdk/basics-settings.html). Using -1 will give an infinite timeout.

**Reply by yehen12** [reply]
I used -1 to give an infinite timeout twenty minutes age , but it still working in converting，it seems the conversion has something wrong cause my excel file only has 5KB size

**Reply by jansenbe** [reply]
@yehen12 : that's not expected...the core SDK is just calling the needed Graph API, so I don't think this is related to the SDK itself. Maybe try with some other files first (e.g. a basic word doc to PDF) to see how that works and then experiment again with Excel files?

**Reply by yehen12** [reply]
If I set PnPCore:Environment with 'China', can it be worked in 21V env? Is Any else will be set?

**Reply by jansenbe** [reply]
@yehen12 : yes for China cloud you need to set that value. Might however be that that MS Graph feature is not available in the China environment. If you do a Fiddler trace you'll learn more on what goes on across the wire.

**Reply by yehen12** [reply]
Thanks a lot for your reply.
I find the difference:
For China 21V Env, Graph Api Domain : https://microsoftgraph.chinacloudapi.cn/
For Global Env : https://graph.microsoft.com/

**Reply by jansenbe** [reply]
@yehen12 : correct, and when you configure PnP Core for China then we'll use that Graph environment. See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/CloudManager.cs#L26-L39
